common/esxi_set_firewall_rule.yml

```diff
@@ -0,0 +1,81 @@
+# Copyright 2023 VMware, Inc.
+# SPDX-License-Identifier: BSD-2-Clause
+---
+# Enable or disable ESXi host firewall rule
+# Parameters:
+#   rule_name: The firewall rule name on ESXi host.
+#   rule_enabled: Whether to enable the firewall rule on ESXi host.
+#   rule_allowed_all: Whether to allow all hosts.
+#   rule_allowed_ip(optional): A list of allowed IP addresses.
+#   rule_allowed_networks(optional): A list of allowed IP networks.
+
+- name: "Check parameters for ESXi host firewall rule"
+  ansible.builtin.assert:
+    that:
+      - rule_name is defined and rule_name
+      - rule_enabled is defined
+      - rule_enabled | type_debug == 'bool'
+    fail_msg: "At least one of parameters 'rule_name' or 'rule_enabled' is incorrect"
+
+- name: "Check parameter for enabling ESXi host firewall rule '{{ rule_name }}'"
+  ansible.builtin.assert:
+    that:
+      - rule_allowed_all is defined
```

Comment:
I think here `rule_allowed_all` can be set according to the parameters `rule_allowed_ip` or `rule_allowed_networks`, if they're set then it's false, if they're not set then it's true. Also these 2 optional paramter value can also be checked.

Reply:
Updated.